scipy/stats/_continuous_distns.py

```diff
@@ -5856,38 +5873,42 @@ def _rvs(self, df, nc, size=None, random_state=None):
 
     def _logpdf(self, x, df, nc):
         cond = np.ones_like(x, dtype=bool) & (nc != 0)
-        return _lazywhere(cond, (x, df, nc), f=_ncx2_log_pdf, f2=chi2.logpdf)
+        return _lazywhere(cond, (x, df, nc), f=_ncx2_log_pdf,
+                          f2=lambda x, df, _: chi2._logpdf(x, df))
```

Comment:
Previously the use of `f2=chi2.logpdf` was a hack. They should have been using `f2=chi2._logpdf` since there's no reason to go through all the `rv_continuous` overhead. But we can't do that because it passes three parameters: `(x, df, nc)`, so we need to define a function to accept three parameters but not use `nc`, since it's zero. I say it was a hack before because it was passing the `nc` parameter - which was always 0 - into the `loc` parameter of `logpdf`. (Same for all the others below.)